admin/class-boldgrid-backup-admin-auto-updates.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * File: class-boldgrid-backup-admin-auto-updates.php
+ *
+ * @link       https://www.boldgrid.com
+ * @since      SINCEVERSION
+ *
+ * @package    Boldgrid_Backup
+ * @subpackage Boldgrid_Backup/admin
+ * @copyright  BoldGrid
+ * @version    SINCEVERSION
+ * @author     BoldGrid <support@boldgrid.com>
+ */
+
+use Boldgrid\Library\Library\Plugin\Plugins;
+
+/**
+ * Class: Boldgrid_Backup_Admin_Auto_Updates.
+ *
+ * @since 1.2
+ */
+class Boldgrid_Backup_Admin_Auto_Updates {
+	/**
+	 * Settings.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	public $settings;
+
+	/**
+	 * Active Plugins.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	public $plugins = array();
+
+	/**
+	 * Themes.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	public $themes = array();
+
+	/**
+	 * Active Plugins.
+	 *
+	 * @since SINCEVERSION
+	 * @var Boldgrid_Backup_Admin_Core
+	 */
+	public $core;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function __construct() {
+
+		$this->set_settings();
+		$plugins       = new \Boldgrid\Library\Library\Plugin\Plugins();
+		$this->plugins = $plugins->getAllPlugins();
+		$this->themes  = new \Boldgrid\Library\Library\Theme\Themes();
+
+		add_filter( 'automatic_updater_disabled', '__return_false' );
+		add_filter( 'auto_update_plugin', array( $this, 'auto_update_plugins' ), 10, 2 );
+		add_filter( 'auto_update_themes', array( $this, 'auto_update_themes' ), 10, 2 );
+
+		add_action( 'admin_init', array( $this, 'auto_update_core' ) );
+	}
+
+	/**
+	 * Is Premium Done.
+	 *
+	 * @since SINCEVERSION
+	 * @return bool
+	 */
+	public function is_premium_done() {
```

Comment:
I remember trying to use the $core->config->is_premium_done() function initially, but had problems. I think it had something to do with the config object not being initialized right when the Auto Updates class called it. Especially since this needs to be able to be called by wordpress during automatic updates, which are run via wp-cron events

Reply:
I figured out how to just get rid of this all together.